sample/src/main/java/com/bukalapak/urlrouter/sample/MainActivity.kt

```diff
@@ -32,42 +34,183 @@ class MainActivity : AppCompatActivity() {
+        firstMap.add(RouterMap(listOf("/product/<product_id:[a-z0-9]+>-*")) {
             val productId = it.variables.getString("product_id")
             displayResult("Open product detail page $productId")
+        })
+
+        router.preMap(listOf("*://", ""),
+                listOf("<subdomain:\\[a-z]+>.mysite.com", "mysite.com"),
+                listOf("/*", "", ":<port:[0-9]+>/*", ":<port:[0-9]+>"),
+                firstMap) {
+            val subdomain = it.variables.getString("subdomain")
+            if (subdomain == "www") {
+                var url = it.url
+                if (!Pattern.matches("\\w+://.+", it.url)) {
+                    url = "https://" + url
+                }
+                Uri.parse(url).path // Continue routing
+            } else {
+                displayResult("Launch intent: " + it.url)
+                null // Don't continue routing
+            }
         }
+
+        val secondMap = mutableListOf<RouterMap>()
+
+        // https://second.mysite.com/about
+        secondMap.add(RouterMap(listOf("/about")) {
+            displayResult("Open second about page")
+        })
+
+        // https://second.mysite.com/promo/tas-keren-pria
+        secondMap.add(RouterMap(listOf("/promo/*")) {
+            displayResult("Open second promo page")
+        })
+
+        // https://second.mysite.com/promo/tas-keren-pria/discounted
+        secondMap.add(RouterMap(listOf("/promo/*/discounted")) {
+            displayResult("Open second discounted promo page")
+        })
+
+        // https://second.mysite.com/register?referrer=anonymous
+        secondMap.add(RouterMap(listOf("/register")) {
+            val referrer = it.queries.getString("referrer")
+            displayResult("Open second registration page with referrer $referrer")
+        })
+
+        // https://second.mysite.com/transaction/981239/view
+        secondMap.add(RouterMap(listOf("/transaction/<transaction_id>/view")) {
+            val transactionId = it.variables.getLong("transaction_id")
+            displayResult("Open second transaction detail page $transactionId")
+        })
+
+        // https://second.mysite.com/product/kj9fd8-tas-paling-keren-masa-kini
+        secondMap.add(RouterMap(listOf("/product/<product_id:[a-z0-9]+>-*")) {
+            val productId = it.variables.getString("product_id")
+            displayResult("Open second product detail page $productId")
+        })
+
+        router.preMap(listOf("*://", ""),
+                listOf("second.mysite.com"),
+                listOf("/*", "", ":<port:[0-9]+>/*", ":<port:[0-9]+>"),
+                secondMap) {
+            var url = it.url
+            if (!Pattern.matches("\\w+://.+", it.url)) {
+                url = "https://" + url
+            }
+            Uri.parse(url).path // Continue routing
+        }
+
+        router.preMap("*://third.mysite.com/*", {
+            Uri.parse(it.url).path // Continue routing
```

Comment:
hmm ini dijadiin default value aja kali ya, ga semua orang butuh ngeintercept kaya gini, malah kebanyakan user selalu continue routing

Reply:
yg ini jadi Uri.parse(it.url).path nya dihandle library aja mas?

di implementasinya DeeplinkRouter ada logic nambahin https, itu bawa ke library juga?

Reply:
kalau dijadiin default continue routing handle yg ga continue nya gimana ya mas?

Reply:
eh engga, jadi ada defaultnya: `Uri.parse(it.url).path`, tapi di deeplinkrouter kan dicustom, itu juga disediain kalo pgn

Reply:
ga continue kalo g ada path yg match